java/src/test/java/com/ncc/aif/NistExamplesAndValidationTest.java

```diff
@@ -735,6 +744,51 @@ void valid() {
                 utils.testValid("LinkAssertion.valid");
             }
         }
+         
+        @Nested
+        class Prototype {
+            @Test
+            void invalidPrototypeOfMultipleClusters() {
+                // create second cluster with same prototype
+                Resource entity1 = makeEntity(model, utils.getEntityUri(), system);
+                markJustification(utils.addType(entity1, LDCOntology.PER), utils.makeValidJustification());
+                makeClusterWithPrototype(model, utils.getClusterUri(), entity1, false, system);
+                makeClusterWithPrototype(model, utils.getClusterUri(), entity1, false, system);
+    
+                utils.expect(ShaclShapes.MultiClusterPrototypeShape, SH.MaxCountConstraintComponent, null);
+                utils.testInvalid("Prototype.invalid: prototype of multiple clusters");
+            }
+            
+            @Test
+            void invalidPrototypeMemberOfOther() {
+                // create sample entities
+                ImmutablePair<Resource, Resource> aPair = utils.makeValidNistEntity(
+                    LDCOntology.PER);
+                final Resource entity2 = aPair.getKey();
+                
+                // create two clusters with different prototypes
+                markAsPossibleClusterMember(model, entity2, entityCluster, 1d, system);
+                
+                utils.expect(ShaclShapes.PrototypeShape, SH.SPARQLConstraintComponent, ShaclShapes.NonClusterPrototypeMemberShape);
+                utils.testInvalid("Prototype.invalid: prototype member of other cluster");
+            }
+        }
+
+        @Nested
+        class Handle {
+            @Test
+            void invalidHandle() {
+                markHandle(entityCluster, "handle");
+                utils.expect(ShaclShapes.PreventHandleOnCluster, SH.NotConstraintComponent, null);
+                utils.testInvalid("Prototype.invalid: prototype of multiple clusters");
```

Comment:
Shouldn't the invalid test message say something similar to "Handle.invalid: handle not allowed on cluster"?